src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs

```diff
@@ -944,27 +944,35 @@ checkInductiveDef InductiveDef {..} = do
 
     checkRecord :: RhsRecord 'Parsed -> Sem r (RhsRecord 'Scoped)
     checkRecord RhsRecord {..} = do
-      fields' <- checkFields _rhsRecordFields
+      fields' <- checkRecordStatements _rhsRecordStatements
       return
         RhsRecord
-          { _rhsRecordFields = fields',
+          { _rhsRecordStatements = fields',
             _rhsRecordDelim
           }
       where
-        checkFields :: [RecordField 'Parsed] -> Sem r [RecordField 'Scoped]
-        checkFields = \case
+        checkRecordStatements :: [RecordStatement 'Parsed] -> Sem r [RecordStatement 'Scoped]
+        checkRecordStatements = \case
           [] -> return []
-          RecordField {..} : fs -> do
-            type' <- checkParseExpressionAtoms _fieldType
-            withLocalScope $ do
-              name' <- bindVariableSymbol _fieldName
-              let f =
-                    RecordField
-                      { _fieldType = type',
-                        _fieldName = name',
-                        ..
-                      }
-              (f :) <$> checkFields fs
+          f : fs -> case f of
+            RecordStatementOperator d ->
+              (RecordStatementOperator d :) <$> checkRecordStatements fs
+            RecordStatementField d -> do
+              d' <- checkField d
+              (RecordStatementField d' :) <$> checkRecordStatements fs
+
+        checkField :: RecordField 'Parsed -> Sem r (RecordField 'Scoped)
+        checkField RecordField {..} = do
+          type' <- checkParseExpressionAtoms _fieldType
+          -- TODO local scope??
+          withLocalScope $ do
```

Comment:
I don't understand the comment. We have `withLocalScope` here. Either the comment should be made more clear or removed